### PR TITLE
Lingtober: Lingmania: The lingaling: Powerling: Lingcreep

### PR DESCRIPTION
**Pull request by Geevies**
The ling is ass. It's just. Ass. Butts. It isn't scary. Whap it and cuff it, toss it in solitary and smack its horribly deformed ass. Now they can tear through walls. Nice.

* Forming armblades and flesh riot shields as ling now break cuffs and straitjackets.
* Changing into a lesser form now uncuffs you
* You can use an armblade is a crowbar, to do crowbar based activities.
* The armblade can be used to slice and dice through non-reinforced walls and girders.

### CODE REVIEW

**Comment by Geevies** [comment]
!review

**Comment by Butterrobber202** [comment]
this really tickles my flesh probes

File: code/game/gamemodes/changeling/changeling_items.dm [hunk 1]
```diff
@@ -47,6 +47,9 @@
 			host.drop_from_inventory(src)
 		QDEL_IN(src, 1)
 
+/obj/item/weapon/melee/arm_blade/iscrowbar()
+	return TRUE
+
```
Comment (fernerr) [inline code/game/gamemodes/changeling/changeling_items.dm]:
```suggestion

/obj/item/weapon/melee/arm_blade/resolve_attackby(atom/A, mob/living/user, var/click_parameters)
	if(istype(A,/turf/simulated/floor) && user.a_intent != I_HELP)
		return
	else
		..()
```

Comment (fernerr) [reply code/game/gamemodes/changeling/changeling_items.dm]: Makes crowbarring tiles with the armblade only work on help, to make fights look less silly.

File: code/game/gamemodes/changeling/changeling_powers.dm [hunk 2]
```diff
@@ -419,6 +419,10 @@ var/global/list/possible_changeling_IDs = list("Alpha","Beta","Gamma","Delta","E
 		to_chat(src, "<span class='warning'>We cannot perform this ability in this form!</span>")
 		return
 
+	if(H.handcuffed)
+		qdel(H.handcuffed)
+		H.handcuffed = null
```
Comment (skull132) [inline code/game/gamemodes/changeling/changeling_powers.dm]:
```suggestion
		var/cuffs = H.handcuffed
		H.u_equip(H.handcuffed)
		qdel(cuffs)
```

You're missing update inventory calls here. So just use `u_equip` tbh.

File: code/game/gamemodes/changeling/changeling_powers.dm [hunk 3]
```diff
@@ -972,11 +976,24 @@ var/list/datum/absorbed_dna/hivemind_bank = list()
...
 
-	var/mob/living/M = src
+	var/mob/living/carbon/M = src
 
 	if(M.l_hand && M.r_hand)
 		to_chat(M, "<span class='danger'>Your hands are full.</span>")
 		return
+	
+	if(M.handcuffed)
+		qdel(M.handcuffed)
```
Comment (skull132) [inline code/game/gamemodes/changeling/changeling_powers.dm]: Same here as above.

File: code/game/gamemodes/changeling/changeling_powers.dm [hunk 4]
```diff
@@ -995,12 +1012,25 @@ var/list/datum/absorbed_dna/hivemind_bank = list()
...
 
-	var/mob/living/M = src
+	var/mob/living/carbon/M = src
 
 	if(M.l_hand && M.r_hand)
 		to_chat(M, "<span class='danger'>Your hands are full.</span>")
 		return
 
+	if(M.handcuffed)
+		qdel(M.handcuffed)
```
Comment (skull132) [inline code/game/gamemodes/changeling/changeling_powers.dm]: same here as above.

File: code/game/objects/structures/girders.dm [hunk 5]
```diff
@@ -111,6 +111,13 @@
 		to_chat(user, "<span class='notice'>You drill through the girder!</span>")
 		dismantle()
 
+	else if(istype(W, /obj/item/weapon/melee/arm_blade/))
+		to_chat(user, "<span class='notice'>Now slicing apart the girder...</span>")
+		if(do_after(user,150))
+			if(!src) return
```
Comment (skull132) [inline code/game/objects/structures/girders.dm]:
```suggestion
			if(!src)
				return
```

File: code/game/gamemodes/changeling/changeling_powers.dm [hunk 6]
```diff
@@ -972,11 +977,25 @@ var/list/datum/absorbed_dna/hivemind_bank = list()
...
+		M.u_equip(M.handcuffed)
+		qdel(cuffs)
+
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		if(istype(H.wear_suit, /obj/item/clothing/suit/straight_jacket))
+			src.visible_message("<span class='danger'>[H] tears through the [H.wear_suit] with their grotesque arm blades!</span>",
+								"<span class='danger'>We tear through the [H.wear_suit] with our arm blades!</span>",
+								"<span class='danger'>You hear cloth ripping and tearing!</span>")
+			qdel(H.wear_suit)
```
Comment (skull132) [inline code/game/gamemodes/changeling/changeling_powers.dm]:
```suggestion
			QDEL_IN(H.wear_suit, 0)
			H.unEquip(H.wear_suit, use_force = TRUE)
```

File: code/game/gamemodes/changeling/changeling_powers.dm [hunk 7]
```diff
@@ -995,12 +1014,26 @@ var/list/datum/absorbed_dna/hivemind_bank = list()
...
+		M.u_equip(M.handcuffed)
+		qdel(cuffs)
+
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		if(istype(H.wear_suit, /obj/item/clothing/suit/straight_jacket))
+			src.visible_message("<span class='danger'>[H] tears through the [H.wear_suit] with their grotesque shield!</span>",
+							"<span class='danger'>We tear through the [H.wear_suit] with our newly formed shield!</span>",
+							"<span class='danger'>You hear cloth ripping and tearing!</span>")
+			qdel(H.wear_suit)
```
Comment (skull132) [inline code/game/gamemodes/changeling/changeling_powers.dm]:
```suggestion
			QDEL_IN(H.wear_suit, 0)
			H.unEquip(H.wear_suit, use_force = TRUE)
```

File: code/game/gamemodes/changeling/changeling_powers.dm [hunk 8]
```diff
@@ -995,12 +1014,26 @@ var/list/datum/absorbed_dna/hivemind_bank = list()
...
+		qdel(cuffs)
+
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		if(istype(H.wear_suit, /obj/item/clothing/suit/straight_jacket))
+			src.visible_message("<span class='danger'>[H] tears through the [H.wear_suit] with their grotesque shield!</span>",
+							"<span class='danger'>We tear through the [H.wear_suit] with our newly formed shield!</span>",
+							"<span class='danger'>You hear cloth ripping and tearing!</span>")
+			QDEL_IN(H.wear_suit, 0)
+			H.unEquip(H.wear_suit, use_force = TRUE)
```
Comment (skull132) [inline code/game/gamemodes/changeling/changeling_powers.dm]:
```suggestion
			H.unEquip(H.wear_suit, force = TRUE)
```

File: code/game/gamemodes/changeling/changeling_powers.dm [hunk 9]
```diff
@@ -972,11 +977,25 @@ var/list/datum/absorbed_dna/hivemind_bank = list()
...
+		qdel(cuffs)
+
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		if(istype(H.wear_suit, /obj/item/clothing/suit/straight_jacket))
+			src.visible_message("<span class='danger'>[H] tears through the [H.wear_suit] with their grotesque arm blades!</span>",
+								"<span class='danger'>We tear through the [H.wear_suit] with our arm blades!</span>",
+								"<span class='danger'>You hear cloth ripping and tearing!</span>")
+			QDEL_IN(H.wear_suit, 0)
+			H.unEquip(H.wear_suit, use_force = TRUE)
```
Comment (skull132) [inline code/game/gamemodes/changeling/changeling_powers.dm]:
```suggestion
			H.unEquip(H.wear_suit, force = TRUE)
```

**Pull request merged**